### PR TITLE
FINERACT-1971: Versioning and proper isolation for loan transactions

### PR DESCRIPTION
**Pull request by adamsaghy**
## Description

- Add versioning for Loan transactions (hence they got updated time to time, ex: reversal, loan balance, etc.)
- Add proper isolation to avoid situation where Tx1 reads changes of Tx2 

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/core/config/Resilience4JProperties.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import io.github.resilience4j.common.retry.configuration.CommonRetryConfigurationProperties;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@Getter
+@Setter
+@ConfigurationProperties(prefix = "resilience4j")
+public class Resilience4JProperties {
```
Comment (vidakovic) [inline fineract-core/src/main/java/org/apache/fineract/infrastructure/core/config/Resilience4JProperties.java]: Is this intended to overlap with Resilience4j's "namespace"? If these are Fineract specific properties applied to Resilience4j then I'd prefix this rather with "fineract.resilience4j"

Comment (adamsaghy) [reply fineract-core/src/main/java/org/apache/fineract/infrastructure/core/config/Resilience4JProperties.java]: yes, it was intended. Priorly Retry annotation was used with a specific name and it was loading the configurations from the application properties. I had to change to use programatic approach to be able to have more fine-grained configuration. 

I would be hesitant to use `fineract.resilience4j.*` for programatic retry logic and `resilience4j.*` for declarative.

Thoughts?

Comment (adamsaghy) [reply fineract-core/src/main/java/org/apache/fineract/infrastructure/core/config/Resilience4JProperties.java]: You convinced me :)

**Review by marta-jankovics — APPROVED** [review]
Looks good to me. After the build has fixed, it can be merged.

**Pull request closed without merging**